### DISCUSSION TITLE
[OKD 4.12] Upgrade from 4.11.  Failure during rpm-ostree rebase

### DISCUSSION THREAD

**Discussion by BonzTM** [okd-project/okd#1503] [General]
**Describe the bug**
```
E0209 23:34:39.857504 2274 writer.go:200] Marking Degraded due to: failed to update OS to quay.io/openshift/okd-content@sha256:ae04fc72f2532d27d75c628cd242d0ab3bd9ea8001e4333153ffc5db2bfb6a79 : error running rpm-ostree rebase --experimental ostree-unverified-registry:quay.io/openshift/okd-content@sha256:ae04fc72f2532d27d75c628cd242d0ab3bd9ea8001e4333153ffc5db2bfb6a79: error: No enabled repositories
: exit status 1
```

When attempting to run `rpm-ostree rebase` manually with the `-C` option, I received a slightly different error
`error: Packages not found: glusterfs, glusterfs-fuse`

Both master and worker Machine Config Operator daemons are spitting this error

All repos are disabled
```
[core@okd-master-1 yum.repos.d]$ grep enabled=1 /etc/yum.repos.d/*
[core@okd-master-1 yum.repos.d]$
```

**Version**
Attempting an update from `4.11.0-0.okd-2023-01-14-152430` to `4.12.0-0.okd-2023-02-04-212953`

**Must Gather**
https://drive.google.com/file/d/18lQRiOszSPmkISKNAWMKpzo2Jopspv8o/view?usp=sharing

**Comment by vrutkovs** [comment]
Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

**Reply by BonzTM** [reply]
> Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

A link to my must gather has been added above

**Comment by BonzTM** [comment]
This discussion can be considered solved by #1483 

Apologies for the inconvenience.  The cluster was initialized at 4.7 and carried through to 4.11, so glusterfs packages needed to be manually uninstalled.  I hope a solution to do these things during the upgrade is implemented for future-type upgrades
